hedera-mirror-rosetta/app/services/construction/construction_service_test.go

```diff
@@ -0,0 +1,457 @@
+	decodedPublicKeyBytes, e2 := hex.DecodeString(publicKeyBytes)
+	decodedSignatureBytes, e3 := hex.DecodeString(signatureBytes)
+
+	if e1 != nil || e2 != nil || e3 != nil {
+		return nil
+	}
+
+	return &types.ConstructionCombineRequest{
+		NetworkIdentifier:   networkIdentifier(),
+		UnsignedTransaction: unsignedTxHash,
+		Signatures: []*types.Signature{
+			{
+				SigningPayload: &types.SigningPayload{
+					AccountIdentifier: &types.AccountIdentifier{
+						Address:  "0.0.123352",
+						Metadata: nil,
+					},
+					Bytes:         decodedSigningPayloadBytes,
+					SignatureType: "ed25519",
+				},
+				PublicKey: &types.PublicKey{
+					Bytes:     decodedPublicKeyBytes,
+					CurveType: "edwards25519",
+				},
+				SignatureType: "ed25519",
+				Bytes:         decodedSignatureBytes,
+			},
+		},
+	}
+}
+
+func dummyOperation(index int64, transferType, account, amount string) *types.Operation {
+	return &types.Operation{
+		OperationIdentifier: &types.OperationIdentifier{
+			Index: index,
+		},
+		Type: transferType,
+		Account: &types.AccountIdentifier{
+			Address: account,
+		},
+		Amount: &types.Amount{
+			Value: amount,
+			Currency: &types.Currency{
+				Symbol:   "HBAR",
+				Decimals: 8,
+				Metadata: map[string]interface{}{
+					"issuer": "Hedera",
+				},
+			},
+		},
+	}
+}
+
+func networkIdentifier() *types.NetworkIdentifier {
+	return &types.NetworkIdentifier{
+		Blockchain: "SomeBlockchain",
+		Network:    "SomeNetwork",
+		SubNetworkIdentifier: &types.SubNetworkIdentifier{
+			Network: "SomeSubNetwork",
+		},
+	}
+}
+
+func dummyConstructionHashRequest(txHash string) *types.ConstructionHashRequest {
+	return &types.ConstructionHashRequest{
+		NetworkIdentifier: networkIdentifier(),
+		SignedTransaction: txHash,
+	}
+}
+
+func dummyConstructionParseRequest(txHash string, signed bool) *types.ConstructionParseRequest {
+	return &types.ConstructionParseRequest{
+		NetworkIdentifier: networkIdentifier(),
+		Signed:            signed,
+		Transaction:       txHash,
+	}
+}
+
+func dummyPayloadsRequest(operations []*types.Operation) *types.ConstructionPayloadsRequest {
+	return &types.ConstructionPayloadsRequest{
+		Operations: operations,
+	}
+}
+
+func TestNewAccountAPIService(t *testing.T) {
+	assert.IsType(t, &ConstructionAPIService{}, NewConstructionAPIService())
+}
+
+func TestNewConstructionAPIService(t *testing.T) {
+	constructionService := NewConstructionAPIService()
+	assert.IsType(t, &ConstructionAPIService{}, constructionService)
+}
+
+func TestConstructionCombine(t *testing.T) {
+	// given:
+	expectedConstructionCombineResponse := &types.ConstructionCombineResponse{
+		SignedTransaction: validTxHash,
+	}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, dummyConstructionCombineRequest())
+
+	// then:
+	assert.Equal(t, expectedConstructionCombineResponse, res)
+	assert.Nil(t, e)
+}
+
+func TestConstructionCombineThrowsWhenNoSingleSignature(t *testing.T) {
```

Comment:
My guess is that in the real use-case, if Signatures is empty array it won't get to the Construction Service as it won't go through the checks performed by the Rosetta SDK.

I think that it would be better to check with more than one signature rather than none. In that case the name of the test should be changed as-well